File: leetcode_ly/java/List/RemoveDuplicatesFromSortedListTwo.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if(head == null || head.next == null) return head;
        ListNode ptr = new ListNode(0);
        ptr.next = head;
        ListNode copy = ptr;
        ListNode pre = head;
        ListNode pos = head.next;
        while(pos != null){
            if(pos.val != pre.val){
                ptr = ptr.next;
                pre = pre.next;
                pos = pos.next;
            }
            else{
                while(pos != null && pos.val == pre.val){
                    pos = pos.next;
                }
                ptr.next = pos;
                if(pos != null){
                    pre = pos;
                    pos = pos.next;
                }
            }
        }
        //ptr.next = null;
        return copy.next;
    }
}

File: leetcode_ly/c++/List/RemoveDuplicatesFromSortedListTwo.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if(!head || !head->next) return head;
        ListNode* node = new ListNode(-1);
        node->next = head;
        ListNode* low = node;
        ListNode* high = head;
        while(high && high->next){
            if(high->val == high->next->val){
                int value = high->val;
                while(high->val == value){
                    high = high->next;
                    if(!high) break;
                }
                low->next = high;
            }
            else{
                high = high->next;
                low = low->next;
            }
        }
        return node->next;
    }
};